sources/xfscli/src/bin/xfs_admin.rs

```diff
@@ -0,0 +1,239 @@
+    #[argh(switch, short = 'e')]
+    extent_flag: bool,
+
+    /// specifies that the filesystem image to be processed is stored in a regular file at device
+    #[argh(switch, short = 'f')]
+    filesystem_image: bool,
+
+    /// prints the current filesystem label
+    #[argh(switch, short = 'l')]
+    filesystem_label: bool,
+
+    /// prints the current filesystem UUID
+    #[argh(switch, short = 'u')]
+    filesystem_uuid: bool,
+
+    /// set the filesystem label to label
+    #[argh(option, short = 'L')]
+    label: Option<String>,
+
+    /// enable (1) or disable (0) lazy-counters in the filesystem
+    #[argh(option, short = 'c')]
+    lazy_counters: Option<usize>,
+
+    /// enables version 2 log format
+    #[argh(switch, short = 'j')]
+    log_version_2: bool,
+
+    /// filesystem's external log location
+    #[argh(option)]
+    log_location: Option<String>,
+
+    /// enable 32bit project identifier support
+    #[argh(switch, short = 'p')]
+    project_identifier_32_bit: bool,
+
+    /// specifies the device special file where the filesystem's realtime section resides
+    #[argh(option, short = 'r')]
+    realtime_device: Option<String>,
+
+    /// set the UUID of the filesystem to uuid
+    #[argh(option, short = 'U')]
+    uuid: Option<String>,
+
+    /// add or remove features on an existing V5 filesystem
+    #[argh(option, short = 'O')]
+    v5_feature: Option<String>,
+
+    /// prints the version number and exits.
+    #[argh(switch, short = 'V')]
+    version: bool,
+
+    /// block-device
+    #[argh(positional)]
+    device: Option<String>,
+}
+
+impl Args {
+    // Filesystem should not be mounted with args in this function
+    // For more reference check require_offline variable in
+    // https://git.kernel.org/pub/scm/fs/xfs/xfsprogs-dev.git/tree/db/xfs_admin.sh
+    fn is_required_offline(&self) -> bool {
+        self.extent_flag
+            || self.filesystem_image
+            || self.log_version_2
+            || self.project_identifier_32_bit
+            || self.uuid.is_some()
+            || self.lazy_counters.is_some()
+            || self.v5_feature.is_some()
+            || self.realtime_device.is_some()
+    }
+}
+
+fn main() {
+    let args: Args = argh::from_env();
+    match run(args) {
+        Ok(exit_code) => exit(exit_code),
+        Err(e) => {
+            eprintln!("{}", e);
+            exit(e.exit_code());
+        }
+    }
+}
+
+fn run(args: Args) -> Result<i32> {
+    // Return the version for arg version as true
+    if args.version {
+        return Ok(query_xfs_db(vec!["-p", "xfs_admin", "-V"]));
+    }
+
+    let device = &args.device.as_ref().context(error::ParseTargetSnafu)?;
+
+    // Collect all the arguments for XFS_DB, XFS_IO and XFS_REPAIR commands
+    // by parsing the CLI inputs.
+    let mut db_opts: Vec<&str> = vec![];
+    let mut io_opts: Vec<&str> = vec![];
+    let mut repair_opts: Vec<&str> = vec![];
+
+    let mut lazycount_arg = "lazycount=".to_string();
```

Comment:
These mutable strings are a little unusual to me, but I guess it's like this so that you can mutate the value while having it guaranteed to live long enough to keep an `&str` reference to it?

Reply:
yes, we need these values to live long enough in order to pass as an argument.